hs_file_types/models/netcdf.py

```diff
@@ -0,0 +1,715 @@
+                if nc_dump.get_nc_dump_string_by_ncdump(temp_file):
+                    dump_str = nc_dump.get_nc_dump_string_by_ncdump(temp_file)
+                else:
+                    dump_str = nc_dump.get_nc_dump_string(temp_file)
+
+                if dump_str:
+                    # refine dump_str first line
+                    first_line = list('netcdf {0} '.format(nc_file_name))
+                    first_line_index = dump_str.index('{')
+                    dump_str_list = first_line + list(dump_str)[first_line_index:]
+                    dump_str = "".join(dump_str_list)
+                    dump_file_name = nc_file_name + '_header_info.txt'
+                    dump_file = os.path.join(temp_dir, dump_file_name)
+                    with open(dump_file, 'w') as dump_file_obj:
+                        dump_file_obj.write(dump_str)
+
+                    files_to_add_to_resource.append(dump_file)
+
+                with transaction.atomic():
+                    # first delete the netcdf file that we retrieved from irods
+                    # for setting it to netcdf file type
+                    delete_resource_file(resource.short_id, res_file.id, user)
+
+                    # create a netcdf logical file object to be associated with
+                    # resource files
+                    logical_file = cls.create()
+
+                    # by default set the dataset_name attribute of the logical file to the
+                    # name of the file selected to set file type unless the extracted metadata
+                    # has a value for title
+                    dataset_title = res_dublin_core_meta.get('title', None)
+                    if dataset_title is not None:
+                        logical_file.dataset_name = dataset_title
+                    else:
+                        logical_file.dataset_name = nc_file_name
+                    logical_file.save()
+
+                    try:
+                        # create a folder for the raster file type using the base file
+                        # name as the name for the new folder
+                        new_folder_path = 'data/contents/{}'.format(nc_file_name)
+                        # To avoid folder creation failure when there is already matching
+                        # directory path, first check that the folder does not exist
+                        # If folder path exists then change the folder name by adding a number
+                        # to the end
+                        istorage = resource.get_irods_storage()
+                        counter = 0
+                        new_file_name = nc_file_name
+                        while istorage.exists(os.path.join(resource.short_id, new_folder_path)):
+                            new_file_name = nc_file_name + "_{}".format(counter)
+                            new_folder_path = 'data/contents/{}'.format(new_file_name)
+                            counter += 1
```

Comment:
Wouldn't it be better to create a random string as path rather than using resource short id? That way, you don't have to check whether the temp folder already exists.